### PR TITLE
Secure communication capabilities

### PR DESCRIPTION
**Pull request by MiguelazoDS**
# Description 

This PR adds the required changes to connect to a secure server with basic and client authentication.

### CODE REVIEW

File: include/HTTPRequest.hpp [hunk 1]
```diff
@@ -34,12 +34,14 @@ class HTTPRequest final
...
      * @param onError Callback to be called in case of error.
      * @param httpHeaders Headers to be added to the query.
+     * @param secureCommunication Object that provides secure communication.
      */
     void download(
         const URL& url,
         const std::string& fileName,
         std::function<void(const std::string&, const long)> onError = [](auto, auto) {},
-        const std::unordered_set<std::string>& httpHeaders = std::unordered_set<std::string>());
+        const std::unordered_set<std::string>& httpHeaders = std::unordered_set<std::string>(),
```
Comment (diegodfrf) [inline include/HTTPRequest.hpp:43]: Isn't it better ?
```suggestion
        const std::unordered_set<std::string>& httpHeaders = {};
```

File: include/IURLRequest.hpp [hunk 2]
```diff
@@ -128,12 +129,14 @@ class IURLRequest
...
      * @param onError Callback to be called when an error occurs.
      * @param httpHeaders Headers to be added to the query.
+     * @param secureCommunication Secure communication object.
      */
     virtual void download(
         const URL& url,
         const std::string& fileName,
         std::function<void(const std::string&, const long)> onError = [](auto, auto) {},
-        const std::unordered_set<std::string>& httpHeaders = std::unordered_set<std::string>()) = 0;
+        const std::unordered_set<std::string>& httpHeaders = std::unordered_set<std::string>(),
```
Comment (diegodfrf) [inline include/IURLRequest.hpp:138]: Isn't it better ?
```suggestion
        const std::unordered_set<std::string>& httpHeaders = {};
```

File: include/secureCommunication.hpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    std::string m_sslKey;
+    std::string m_basicAuthCreds;
+
+public:
+    /**
+     * @brief Construct a new Secure Communication object.
+     *
+     * @param caRootCertificate CA certificate path.
+     */
+    SecureCommunication(const std::string& caRootCertificate)
```
Comment (diegodfrf) [inline include/secureCommunication.hpp:36]:
```suggestion
    SecureCommunication(const std::string& caRootCertificate):
    m_caRootCertificate{caRootCertificate}
```

**Review by diegodfrf — CHANGES_REQUESTED** [review]

File: include/HTTPRequest.hpp (hunk 1 above)
Comment (MiguelazoDS) [reply include/HTTPRequest.hpp:43]: That was implemented prior to this PR but it's a good catch to use the constructor by default. Changed.

File: include/IURLRequest.hpp (hunk 2 above)
Comment (MiguelazoDS) [reply include/IURLRequest.hpp:138]: Ditto

File: include/secureCommunication.hpp (hunk 3 above)
Comment (MiguelazoDS) [reply include/secureCommunication.hpp:36]: Changed

**Review by diegodfrf — APPROVED** [review]

**Pull request merged**